skunkworks_crow/src/main/java/org/odk/share/views/ui/instance/adapter/InstanceAdapter.java

```diff
@@ -54,14 +58,47 @@ public void onClick(View v) {
             }
         });
         holder.title.setText(cursor.getString(cursor.getColumnIndex(InstanceProviderAPI.InstanceColumns.DISPLAY_NAME)));
-        holder.subtitle.setText(cursor.getString(cursor.getColumnIndex(InstanceProviderAPI.InstanceColumns.DISPLAY_SUBTEXT)));
+
+        long lastStatusChangeDate = getCursor().getLong(getCursor().getColumnIndex(InstanceProviderAPI.InstanceColumns.LAST_STATUS_CHANGE_DATE));
+        String status = getCursor().getString(getCursor().getColumnIndex(InstanceProviderAPI.InstanceColumns.STATUS));
+        String subtext = getDisplaySubtext(context, status, new Date(lastStatusChangeDate));
+
+        holder.subtitle.setText(subtext);
         long id = cursor.getLong(cursor.getColumnIndex(InstanceProviderAPI.InstanceColumns._ID));
         holder.checkBox.setChecked(selectedInstances.contains(id));
         holder.reviewedForms.setVisibility(View.GONE);
         holder.unReviewedForms.setVisibility(View.GONE);
 
     }
 
+    public static String getDisplaySubtext(Context context, String state, Date date) {
+        try {
+            if (state == null) {
+                return new SimpleDateFormat(context.getString(R.string.added_on_date_at_time),
+                        Locale.getDefault()).format(date);
+            } else if (InstanceProviderAPI.STATUS_INCOMPLETE.equalsIgnoreCase(state)) {
+                return new SimpleDateFormat(context.getString(R.string.saved_on_date_at_time),
+                        Locale.getDefault()).format(date);
+            } else if (InstanceProviderAPI.STATUS_COMPLETE.equalsIgnoreCase(state)) {
+                return new SimpleDateFormat(context.getString(R.string.finalized_on_date_at_time),
+                        Locale.getDefault()).format(date);
+            } else if (InstanceProviderAPI.STATUS_SUBMITTED.equalsIgnoreCase(state)) {
+                return new SimpleDateFormat(context.getString(R.string.sent_on_date_at_time),
+                        Locale.getDefault()).format(date);
+            } else if (InstanceProviderAPI.STATUS_SUBMISSION_FAILED.equalsIgnoreCase(state)) {
+                return new SimpleDateFormat(
+                        context.getString(R.string.sending_failed_on_date_at_time),
+                        Locale.getDefault()).format(date);
+            } else {
+                return new SimpleDateFormat(context.getString(R.string.added_on_date_at_time),
+                        Locale.getDefault()).format(date);
+            }
+        } catch (IllegalArgumentException e) {
+            Timber.e(e);
+            return "";
```

Comment:
return `null`.

Reply:
I think returning empty string is similar to null value returned. 
Because if you call textView.setText(null) and textView.setText(""), they both resets the previous string and sets nothing on the textView.